lib/Resolver.js

```diff
@@ -71,11 +71,24 @@ Resolver.prototype.resolve = function resolve(context, path, request, callback)
 		}
 		return callback(null, result.path === false ? false : result.path + (result.query || ""), result);
 	}
-	return this.doResolve("resolve", obj, message, createInnerCallback(onResolved, {
+	return this.doResolveWithErrorLogging("resolve", obj, message, onResolved, {
 		log: writeLog,
 		missing: missing,
 		stack: callback.stack
-	}, null));
+	});
+};
+
+Resolver.prototype.doResolveWithErrorLogging = function doResolveWithErrorLogging(type, request, message, callback, options) {
```

Comment:
I guess further optimization can be done if this is inlined into `resolve`. i. e. `missing` is only needed for the error run if `callback.missing` is unset.